package.json

```diff
@@ -97,14 +99,206 @@
           "type": "boolean",
           "default": true,
           "description": "Should output channel be raised on formatting error."
+        },
+        "zig.zls.enabled": {
+          "scope": "resource",
+          "type": "boolean",
+          "description": "Whether to enable zls",
+          "default": true
+        },
+        "zig.zls.trace.server": {
+          "scope": "window",
+          "type": "string",
+          "enum": [
+            "off",
+            "messages",
+            "verbose"
+          ],
+          "default": "off",
+          "description": "Traces the communication between VS Code and the language server."
+        },
+        "zig.zls.check_for_update": {
```

Comment:
I'm fine with either snake case or camel case but we should make it consistent.

Reply:
We'll have to revamp how we do LSP configuration because of limitations with how it works by default (limitations with name compatibility, required correct order; if the configuration option order is off, we get garbage data).